hibernation-setup-tool.c

```diff
@@ -1644,15 +1644,105 @@ static int handle_systemd_suspend_notification(const char *argv0, const char *wh
     return 1;
 }
 
-static void link_hook(const char *src, const char *dest)
+static bool link_hook(const char *src, const char *dest)
 {
     if (link(src, dest) < 0) {
         if (errno != EEXIST)
-            return log_fatal("Couldn't link %s to %s: %s", src, dest, strerror(errno));
+            return false;
     }
 
     log_info("Notifying systemd of new hooks");
     spawn_and_wait("systemctl", 1, "daemon-reload");
+    return true;
+}
+
+static void optional_params(int argc, char **argv, char **action, char **when, char **dest_dir) {
+    if (argc > 2) {   
+        int opt;
+        while ((opt = getopt(argc, argv, "adw:")) != -1)
+        {
+            switch (opt)
+            {
+                case 'a':
+                    *action = optarg; 
+                    break; 
+                 
+                case 'w':
+                    *when = optarg; 
+                    break;
+
+                case 'd':
+                    *dest_dir = optarg;
+                    break;
+            }
+        }
+    }
+}
+
+static bool ensure_systemd_service_enabled(char *dest_dir){
+    const char *execfn = (const char *)getauxval(AT_EXECFN);
+    const char *usr_sbin_default = "/usr/sbin", *systemd_dir_default = "/lib/systemd/system";
```

Comment:
also usr/lib/systemd/system based on the distro?

Reply:
On Ubuntu /usr/lib and /lib and different directories. On Ubuntu, /usr/lib is a user-based directory to store things that require access to libraries. The /lib folder is the actual place for essential standard libraries. We need to put our services in this directory in order to access the targets we need (hibernate.target) and for systemd to recognize us as a legitimate service. 

After testing on Debian, Rhel and CentOS, the /usr/lib and /lib directories are identical. lib is a symbolic link pointing to /usr/lib. In other words, /lib and /usr/lib redirect go to the same place. 

/lib is preferred because it includes ubuntu and so works on all distros. 

Additional testing: On Debian tool executes without any issues.

However, Rhel has the additional challenges of not having enough space on root "/" partition no matter how much space it's created with. This causes the free space check to prevent tool from running, and if that is removed, it causes insufficient space error upon allocating hibfile.